docs/xocto/model_fields.md

```diff
@@ -0,0 +1,161 @@
+# Model Fields
+
+Custom [Django model fields](https://docs.djangoproject.com/en/dev/ref/models/fields/).
+
+
+## Postgres specific fields
+
+Fields that can only be used with a Postgres database, enhancing the functionality
+provided by [django.contrib.postgres.fields](https://docs.djangoproject.com/en/dev/ref/contrib/postgres/fields/).
+
+### Ranges
+
+Provides [range fields](https://docs.djangoproject.com/en/dev/ref/contrib/postgres/fields/#range-fields) that work
+with the `Range` classes from [xocto.ranges](./ranges.md).
+
+The underlying Range type exposed by the Django ORM is `psycopg2.extras.Range` which is awkward to
+use throughout an application, and requires lots of boilerplate code to work with correctly.
+
+Alternatively, `xocto.ranges` are designed to be used throughout an application. These fields
+make it possible to interact with them directly from the Django ORM.
+
+Migrating from the Django range types to these range types is straightforward and does
+not require any changes to the database schema. The migration file generated will refer
+to the new fields, but since the underlying database type is the same, the migration will
+be a no-op.
+
+The standard Django query operators are almost the same as for the built-in types.
+They accept `xocto.ranges` as arguments, but don't support passing in a tuple of values:
+
+```python
+
+assert SalesPeriod.objects.filter(
+    period__contains=ranges.FiniteDateRange(
+        start=datetime.date(2020, 1, 10),
+        end=datetime.date(2020, 1, 20),
+    )
+).exists()
+
+assert SalesPeriod.objects.filter(
+    period__overlaps=ranges.FiniteDateRange(
+        start=datetime.date(2020, 1, 10),
+        end=datetime.date(2020, 1, 20),
+    )
+).exists()
+
+# ERROR! This will raise a TypeError
+SalesPeriod.objects.filter(period__overlaps=(datetime.date(2020, 1, 10), datetime.date(2020, 1, 20)))
+
+```
+
+#### FiniteDateRangeField
+
+Module: `xocto.fields.postgres.ranges.FiniteDateRangeField`\
+Bounds: `[]`\
+Type: [xocto.ranges.FiniteDateRange](xocto.ranges.FiniteDateRange)
+
+A field that represents an inclusive-inclusive `[]` ranges of dates. The start
+and end of the range are inclusive and must not be `None`.
+
+```python
+import datetime
+from django.db import models
+from xocto import ranges
+from xocto.fields.postgres import ranges as db_ranges
+
+class SalesPeriod(models.Model):
+    ...
+    period = db_ranges.FiniteDateRangeField()
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** Perhaps an example or 2 of using these range fields in constraints

One of the major benefits of Posgres ranges are that they make it a lot easier to implement non-overlapping constraints. Some examples of how to use these range fields for that would be handy. ‏

Reply:
I plan to add a set of useful constraints in a follow up PR, and will document them properly once done